File: android/src/main/java/com/example/honeywell_rfid_reader_android/messages/model/ConnectionStatus.java
package com.example.honeywell_rfid_reader_android.messages.model;

public enum ConnectionStatus {
    DISCONNECTED(0),
    CONNECTED(1);
    private final int value;

    ConnectionStatus(int value) {
        this.value = value;
    }

    public int getStatusCode() {
        return value;
    }
}

File: lib/model/connection_status.dart
enum ConnectionStatus {
  DISCONNECTED(0),
  CONNECTED(1);

  const ConnectionStatus(this.value);

  final int value;

  int getValue() {
    return value;
  }

  static ConnectionStatus fromValue(int value) {
    return ConnectionStatus.values
        .firstWhere((element) => element.value == value);
  }
}

extension ConnectionStatusExtension on ConnectionStatus {
  bool get isConnected => this == ConnectionStatus.CONNECTED;

  bool get isDisconnected => this == ConnectionStatus.DISCONNECTED;
}
